### PR TITLE
Garrison stuff

### PR DESCRIPTION
**Pull request by funjoker**

### CODE REVIEW

File: WowPacketParserModule.V7_0_3_22248/Parsers/GarrisonHandler.cs [hunk 1]
```diff
@@ -192,5 +192,75 @@ public static void HandleGetGarrisonInfoResult(Packet packet)
...
+            packet.ResetBitReader();
+
+            packet.ReadBit("UnkBit");
+        }
+
+        [Parser(Opcode.SMSG_GARRISON_START_MISSION_RESULT)]
+        public static void HandleGarrisonStartMissionResult(Packet packet)
+        {
+            packet.ReadInt32("Result");
+            packet.ReadInt16("UnkInt16"); // increases per Mission
```
Comment (soulfrost) [inline WowPacketParserModule.V7_0_3_22248/Parsers/GarrisonHandler.cs]: dk about blizz name but it's counter of missions started on session

File: WowPacketParserModule.V7_0_3_22248/Parsers/GarrisonHandler.cs [hunk 2]
```diff
@@ -192,5 +192,75 @@ public static void HandleGetGarrisonInfoResult(Packet packet)
...
+            packet.ReadUInt32("UnkUInt32");
+            ReadGarrisonFollower(packet);
+            ReadGarrisonFollower(packet);
+        }
+
+        [Parser(Opcode.CMSG_OPEN_MISSION_NPC)]
+        public static void HandleGarrisonNpcGUID(Packet packet)
+        {
+            packet.ReadPackedGuid128("NpcGUID");
+            packet.ReadInt32E<GarrisonMissionType>("GarrMissionType");
```
Comment (soulfrost) [inline WowPacketParserModule.V7_0_3_22248/Parsers/GarrisonHandler.cs]: this is FollowerTypeID
like this
            Garrison = 1,
            Shipyard = 2,
            Follower = 4

File: WowPacketParserModule.V7_0_3_22248/Parsers/GarrisonHandler.cs [hunk 3]
```diff
@@ -192,5 +219,140 @@ public static void HandleGetGarrisonInfoResult(Packet packet)
...
+            ReadGarrisonMission(packet, "Mission");
+            packet.ReadUInt32("MissionRecId");
+
+            if (ClientVersion.AddedInVersion(ClientVersionBuild.V7_2_0_24015)) // even earlier?
+            {
+                uint count = packet.ReadUInt32("MissionFollowerCount");
+                for (uint i = 0; i < count; i++)
+                {
+                    packet.ReadUInt64("FollowerDBID", i);
+                    packet.ReadUInt32("UnkUint32", i); // MissionCompleteState ?
```
Comment (funjoker) [inline WowPacketParserModule.V7_0_3_22248/Parsers/GarrisonHandler.cs]: MissionCompleteState <-- ?

Comment (soulfrost) [reply WowPacketParserModule.V7_0_3_22248/Parsers/GarrisonHandler.cs]: oh, dk about this one, mostly seen there just zero

Comment (funjoker) [reply WowPacketParserModule.V7_0_3_22248/Parsers/GarrisonHandler.cs]: 0 <-- SUCCESS well i keep it open for now ^^

**Comment by funjoker** [comment]
Makes #370 obsolete

File: WowPacketParserModule.V7_0_3_22248/Parsers/GarrisonHandler.cs [hunk 4]
```diff
@@ -16,24 +16,16 @@ public static void ReadGarrisonMission(Packet packet, params object[] indexes)
             packet.ReadTime("StartTime", indexes);
             packet.ReadUInt32("TravelDuration", indexes);
             packet.ReadUInt32("MissionDuration", indexes);
-            packet.ReadUInt32("MissionState", indexes);
-            packet.ReadUInt32("Unknown1", indexes);
-            packet.ReadUInt32("Unknown2", indexes);
+            packet.ReadUInt32E<GarrisonMissionState>("MissionState", indexes);
+            packet.ReadUInt32("SuccessChance", indexes);
+            packet.ReadUInt32E<GarrisonMissionFlag>("Flags", indexes);
```
Comment (soulfrost) [inline WowPacketParserModule.V7_0_3_22248/Parsers/GarrisonHandler.cs]: are u sure about this field?

File: WowPacketParser/Enums/GarrisonEnums.cs [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+    {
+        None            = 0,
+        DraenorGarrison = 2,
+        OrderHall       = 3,
+        BfA             = 9
+    }
+
+    public enum GarrisonResult : int
+    {
+        SUCCESS                                   = 0,
```
Comment (Shauren) [inline WowPacketParser/Enums/GarrisonEnums.cs]: naaaaaaming again

File: WowPacketParserModule.V7_0_3_22248/Parsers/GarrisonHandler.cs (hunk 4 above)
Comment (funjoker) [reply WowPacketParserModule.V7_0_3_22248/Parsers/GarrisonHandler.cs]: Companion receives same packets and handels this part as flag.. so yeah pretty

File: WowPacketParser/Enums/GarrisonEnums.cs (hunk 5 above)
Comment (funjoker) [reply WowPacketParser/Enums/GarrisonEnums.cs]: REEEEEE ok

**Comment by funjoker** [comment]
Review and merge?

**Pull request merged**